### PR TITLE
Reportal tableau jobtype

### PR DESCRIPTION
**Pull request by chengren311**
This PR adds tableau jobtype for reportal.
Tableau is not real-time: whenever users want to incorporate the newest data into their visualization, they must refresh the extracted data. This tells Tableau to get and show whatever new data has come into the data sources. It's natural for users to want to automatically trigger Tableau refreshes when their jobs are done and their data sources are full of new data. This job is designed to do exactly that - refreshing a Tableau workbook whenever run so users don't have to do so manually. 

Since tableau workbook refresh doesn't require authentication, all users have to do is identify the name of the workbook they'd like to refresh and it will do so.

### CODE REVIEW

File: az-reportal/src/main/java/azkaban/reportal/util/tableau/URLResponse.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  private String _urlContents;
+
+  public URLResponse(final String tableauUrl, final Path path, final String workbook) throws
+      Exception {
+    this._url = new URL(tableauUrl + "/" + path.getPath() + workbook);
+    refreshContents();
+  }
+
+  public void refreshContents() throws IOException {
+    this._urlContents = IOUtils.toString(this._url.openStream(), "UTF-8");
```
[the remark this answers is not in the document]
Comment (chengren311) [reply az-reportal/src/main/java/azkaban/reportal/util/tableau/URLResponse.java]: _urlContents is the response from this._url, so it's a fixed set of value.

File: az-reportal/src/test/java/azkaban/reportal/util/tableau/CountDownTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.Duration;
+import org.junit.Test;
+
+public class CountDownTest {
+
+  @Test
+  public void testMoreTimeRemaining() throws InterruptedException {
```
Comment (jamiesjc) [inline az-reportal/src/test/java/azkaban/reportal/util/tableau/CountDownTest.java]: Unit tests usually finish very soon, within 1-2 seconds. Is it possible to reduce the runtime of this test using some mocking method?

Comment (HappyRay) [reply az-reportal/src/test/java/azkaban/reportal/util/tableau/CountDownTest.java]: A good unit test should finish within milliseconds. 
@jamiesjc
Thanks for paying attention to test running time.

File: az-reportal/src/main/java/azkaban/reportal/util/tableau/URLResponse.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  private Boolean indicates(final String word) {
+    if (this._urlContents == null) {
+      return false;
+    } else {
+      return this._urlContents.contains(word);
+    }
+  }
+
+  public Boolean indicatesSuccess() {
```
Comment (jamiesjc) [inline az-reportal/src/main/java/azkaban/reportal/util/tableau/URLResponse.java]: Why the return type is `Boolean` not `boolean`?

File: az-reportal/src/test/java/azkaban/reportal/util/tableau/CountDownTest.java (hunk 2 above)
Comment (chengren311) [reply az-reportal/src/test/java/azkaban/reportal/util/tableau/CountDownTest.java]: thanks, updated

File: az-reportal/src/main/java/azkaban/reportal/util/tableau/URLResponse.java (hunk 3 above)
Comment (chengren311) [reply az-reportal/src/main/java/azkaban/reportal/util/tableau/URLResponse.java]: updated.

**Comment by chengren311** [comment]
@HappyRay this feature is intended for internal tableau users, so i will compose an internal doc before launch.

**Pull request merged**